include/boost/gil/extension/io/formats/targa/reader_backend.hpp

```diff
@@ -95,11 +95,17 @@ struct reader_backend< Device
         
         _info._descriptor = _io_dev.read_uint8();
 
+        if (_info._image_type == 1 && _info._descriptor != 0)
+        {
+            io_error("Unsupported descriptor for targa file");
+        }
 
-
-        if(_info._bits_per_pixel == 24 && _info._descriptor != 0 ) 
+        if (_info._bits_per_pixel == 24)
```

Comment:
Being a bit nit-picky, but for clarity this should be an `else if`.

Reply:
Good catch.

I also think the following should be test of not the whole descriptor byte but 0-3 bits only as per the TGA spec:

>  Image Descriptor Byte.                                    |
> Bits 3-0 - number of attribute bits associated with each pixel.
> For the Targa 16, this would be 0 or   1.  For the Targa 24, it should be 0.  For Targa 32, it should be 8.   

```
        else if (_info._bits_per_pixel == 24)
        {
            if ((_info._descriptor & 0x0FU) != 0)
            {
```

Am I correct?

Reply:
That makes sense to me. But you should definitely put either some explanation or a reference to the appropriate location in the spec above it, so a casual reader can understand it.

Reply:
@stefanseefeld sure, I will. TBH, there's plenty of places similar references would be strongly welcomed

Reply:
Right, understood. But it's good practice to address those locations that you are looking at anyhow, as that's the moment you know what you are doing... :-)

Reply:
Yes, I agree.